Packs/SecurityIntelligenceServicesFeed/Integrations/SecurityIntelligenceServicesFeed/SecurityIntelligenceServicesFeed.py

```diff
@@ -0,0 +1,540 @@
+        latest_key_list: List[str] = [key_dict.get('Key', '') for key_dict in
+                                      client.request_list_objects(feed_type=feed, start_after=cached_key,
+                                                                  prefix=feed.lower()) if
+                                      key_dict.get('LastModified', datetime.today().date()).date() >= date_from]
+
+        # If get-indicators command called.
+        if limit:
+            if latest_key_list:
+                # Retrieving latest object key for get-indicators.
+                latest_key_list = [latest_key_list[-1]]
+            elif cached_key:
+                # If no latest key found, retrieving key from cache.
+                latest_key_list = [cached_key]
+
+        # Iterating through latest objects found.
+        for key in latest_key_list:
+
+            # Generator, returning object data specified batch wise.
+            feeds_batch = client.build_iterator(feed_type=feed, key=key, limit=limit, search=search,
+                                                batch_size=batch_size)
+            for feed_dicts in feeds_batch:
+                indicators = []
+
+                # Iterating trough each feed dictionary and creating indicators.
+                for feed_dict in feed_dicts:
+                    value = feed_dict.get('value', '')
+                    if value:
+                        indicator_type = INDICATOR_TYPES.get(feed)
+                        feed_dict['type'] = indicator_type
+                        remove_nulls_from_dictionary(feed_dict)
+                        indicators.append({
+                            'value': value,
+                            'type': indicator_type,
+                            'rawJSON': feed_dict,
+                            'fields': indicator_field_mapping(feed, indicator=feed_dict)
+                        })
+                yield indicators
+
+        # Setting last key to context.
+        if latest_key_list and not limit:
+            set_last_key_to_integration_context(feed, latest_key_list[-1])
+
+
+@logger
+def get_indicators_command(client: Client, args: Dict[str, str]) -> CommandResults:
+    """
+    Wrapper for retrieving indicators from the feed to the war-room.
+
+    :param client: Client object.
+    :param args: demisto arguments.
+    :return: CommandResult instance.
+    """
+    # Retrieving command arguments.
+    feed_type = args.get('feed_type', 'domain').lower()
+    feed_type = feed_type if feed_type != 'phishing' else 'phish'
+    limit = args.get('limit', '50')
+    search = args.get('search', '')
+
+    # Validate the provided limit
+    validate_limit(limit)
+
+    # Validate the provided feed type.
+    validate_feeds([feed_type])
+
+    # Retrieving indicators from fetch indicators command.
+    indicators_list: List[Dict[str, Any]] = []
+    for indicators in fetch_indicators_command(client, [feed_type], limit=limit, search=search):
```

Comment:
The `get-indicators` command should be independent and given the same parameters- return the same results.

I cannot test it, but I believe that's not the case here if you're using the ** fetch_indicators_command** which starts to get the indicators after the `cached_key`.

So If you call `fetch_indicators_command` before a fetch was made or after a fetch was made, the last cache will be different and the resulting indicator as well, am I wight?

Reply:
The current approach is to fetch the last uploaded S3 bucket object for an indicator type and show limited (at max 1000) indicators. 
The cache is only used to limit the search for bucket objects, whether the cache is there or not, always the last uploaded bucket object will only be fetched. Our understanding from docs was that this command is to see limited/sample indicators in the war-room.

Reply:
So if I'll fetch indicators twice while there is no change in the buckets, will both executions return the same output?

Reply:
Yes, It will produce same result.